restcomm/restcomm.sms/src/main/java/org/restcomm/connect/sms/smpp/SmppClientOpsThread.java

```diff
@@ -352,19 +391,11 @@ public void firePduRequestExpired(PduRequest pduRequest) {
 
         @Override
         public PduResponse firePduRequestReceived(PduRequest pduRequest) {
-            // TODO : SMPP request received. Let RestComm know so it calls
-            // coresponding App
-
             PduResponse response = pduRequest.createResponse();
-            //Nexmo keeps sending enquire_link in response that causes
-            //problem with normal PDU request. This tells Restcomm SMPP to do Nothing
-            //with the request
-            if (pduRequest.toString().toLowerCase().contains("enquire_link")) {
-                //logger.info("This is a response to the enquire_link, therefore, do NOTHING ");
-            } else {
 
-                DeliverSm deliverSm = (DeliverSm) pduRequest;
+            if (!pduRequest.toString().toLowerCase().contains("enquire_link")) {
```

Comment:
i would like this condition to be more accurate to do the DeliverSm casting later..